apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -59,11 +59,6 @@ type ContourConfigurationSpec struct {
 	// +kubebuilder:default={disablePermitInsecure: false}
 	HTTPProxy HTTPProxyConfig `json:"httpproxy"`
 
-	// LeaderElection contains leader election parameters.
-	// +optional
-	// +kubebuilder:default={leaseDuration: "15s", renewDeadline: "10s", retryPeriod: "2s", disableLeaderElection: false, configmap: {name: "leader-elect", namespace: "projectcontour"}}
-	LeaderElection LeaderElectionConfig `json:"leaderElection"`
```

Comment:
note: this is technically a breaking change on this v1alpha1 CRD, do we need to worry about it since its isn't widely circulated/used likely?

Reply:
Our [documented policy](https://projectcontour.io/resources/deprecation-policy/) is somewhat unclear here: 

The Alpha row says "Behavior can change any time, Fields can be removed at any time", as compared to the Beta row, which says "Behavior can change any time, fields won’t be removed without a version bump (ie v1alpha1 to v1alpha2)". That implies we could drop a field from the alpha group without revving the API version. However, the policy also indicates we're consistent with the Kubernetes policy, which AFAIK does not allow dropping fields without revving the version, even for alpha APIs.

Pragmatically, I feel like this should be OK, but we should also update/clarify our policy if this is something we're comfortable with.

Reply:
yeah agreed, some of the statements in that doc are at odds, I'll make a PR to update it a bit

Reply:
Maybe we can explicitly say that we follow the policy for beta/GA APIs, but for alpha, we're more lenient in that we allow dropping fields without revving the version? I feel like that'd be a reasonably policy.

Reply:
yep, basically what i was gonna change, something to the effect of "we follow k8s policies, with a caveat for alpha"

Reply:
https://github.com/projectcontour/contour/pull/4173

Reply:
I agree on this one, alpha should mean "we can change anything at any time".